### PR TITLE
Fix panic while chain sync

### PR DESCRIPTION
**Pull request by JekaMas**
Adds some additional info for future and adds a check at the reorg stage if new and current blocks are the same.

Closes https://github.com/kowala-tech/kcoin/issues/768

### CODE REVIEW

File: client/core/blockchain.go [hunk 1]
```diff
@@ -1196,6 +1194,13 @@ func countTransactions(chain []*types.Block) (c int) {
 // to be part of the new canonical chain and accumulates potential missing transactions and post an
 // event about them
 func (bc *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
+	if oldBlock.Number().Cmp(newBlock.Number()) == 0 && bytes.Equal(oldBlock.Hash().Bytes(), newBlock.Hash().Bytes()) {
```
Comment (jmatosp) [inline client/core/blockchain.go]: what was this doing then?

File: client/core/blockchain.go [hunk 2]
```diff
@@ -941,7 +939,7 @@ func writePositionalMetadata(batch kcoindb.Batch, block *types.Block, state *sta
 }
 
 func IsHead(block *types.Block, currentBlock *types.Block) bool {
```
Comment (jmatosp) [inline client/core/blockchain.go]: don't think this name is correct, it just check the blocks are same level

File: client/core/blockchain.go (hunk 1 above)
Comment (JekaMas) [reply client/core/blockchain.go]: Checks are current and received blocks the same. Will refactor a bit to make it more clear.

File: client/core/blockchain.go (hunk 2 above)
Comment (JekaMas) [reply client/core/blockchain.go]: IsParent?

File: client/core/types/block.go [hunk 3]
```diff
@@ -352,6 +353,11 @@ func (b *Block) AsFragments(size int) (*BlockFragments, error) {
 	return NewDataSetFromData(rawBlock, size), nil
 }
 
+// IsEqual returns true if blocks have same hashes and block numbers
```
Comment (jmatosp) [inline client/core/types/block.go]: I'm really surprised ethereum didn't have this. I can only imagine how many comparisons exist in code base

Prob the hash comparison was enough

Comment (JekaMas) [reply client/core/types/block.go]: I was surprised too :)

To be sure for 100% lets check hash and block number.

File: client/core/blockchain.go [hunk 4]
```diff
@@ -940,8 +937,8 @@ func writePositionalMetadata(batch kcoindb.Batch, block *types.Block, state *sta
 	rawdb.WritePreimages(batch, block.NumberU64(), state.Preimages())
 }
 
-func IsHead(block *types.Block, currentBlock *types.Block) bool {
-	return block.ParentHash() != currentBlock.Hash()
+func IsParent(currentBlock *types.Block, block *types.Block) bool {
```
Comment (jmatosp) [inline client/core/blockchain.go]: maybe this would suit better as a receiver on block

IsParent and IsChild

Comment (JekaMas) [reply client/core/blockchain.go]: Ok. I think IsParent is enough.

**Pull request merged**